### PR TITLE
testing framework authentication between master and slave

### PR DESCRIPTION
**Pull request by anshupande**

### CODE REVIEW

File: v2/setup/mesos-master.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+HOMEDIR=$(eval echo "~`whoami`")
+
+AWS_CREDS=""
+if [ ! -z $AWS_ACCESS_KEY ]; then
+    AWS_CREDS=" -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY \
+     -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_KEY "
+fi
+
+sudo docker run --rm \
+    -v ${HOMEDIR}:/data/ $AWS_CREDS behance/docker-aws-s3-downloader \
```
Comment (bryanlatten) [inline v2/setup/mesos-master.sh:14]: is this now in the adobe-platform repo now?

File: v2/setup/mesos-slave.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+HOMEDIR=$(eval echo "~`whoami`")
+
+AWS_CREDS=""
+if [ ! -z $AWS_ACCESS_KEY ]; then
+    AWS_CREDS=" -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY \
+     -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_KEY "
+fi
+
+sudo docker run --rm \
+    -v ${HOMEDIR}:/data/ $AWS_CREDS behance/docker-aws-s3-downloader \
```
Comment (bryanlatten) [inline v2/setup/mesos-slave.sh:14]: same comment as above

File: v2/setup/mesos-master.sh (hunk 1 above)
Comment (anshupande) [inline v2/setup/mesos-master.sh:14]: @bryanlatten If I change this for downloading one secret, I will have to change it for all secrets... I think that should be a separate PR. Let me know your views?

File: v2/setup/mesos-slave.sh (hunk 2 above)
Comment (anshupande) [inline v2/setup/mesos-slave.sh:14]: we can change this in separate PR?

File: v2/setup/mesos-master.sh (hunk 1 above)
Comment (bryanlatten) [inline v2/setup/mesos-master.sh:14]: @anshupande please change it - why would we keep a reference to the behance containers?

Comment (matthewdfuller) [inline v2/setup/mesos-master.sh:14]: @bryanlatten because we pull from Behance in numerous places and the change you're proposing would require an update to .dockercfg (new Docker Hub location) as well as all the images that get pulled down elsewhere. Much better to do in a separate PR.

EDIT: we can do this change in V2 as well.

Comment (matthewdfuller) [inline v2/setup/mesos-master.sh:14]: Specifically, here: https://github.com/adobe-platform/mesos-systemd/blob/master/v2/setup/images.sh

**Comment by anshupande** [comment]
screen shot 2016-03-22 at 12 10 44 pm

**Comment by anshupande** [comment]
deployment successful and mesos_master and mesos_slave started using authentication

**Comment by anshupande** [comment]
@bryanlatten @iderdik @matthewdfuller @fraczles this is ready to be merged...I need thumbs up

**Comment by matthewdfuller** [comment]
We should probably test this in V3 now - is there any issue you can think of with authentication using internal routes?

**Comment by anshupande** [comment]
mesos_slave while starting should have same set of secrets as are with mesos_master...If both are using same secrets we will be able to start the service..else the service will fail to start @matthewdfuller

**Comment by anshupande** [comment]
so internal route should not break anything

**Comment by matthewdfuller** [comment]
Okay cool, so let's merge this in and then work together on a PR for V3 support.

**Comment by matthewdfuller** [comment]
:+1:

**Comment by anshupande** [comment]
ok..i will add secrets for stage/prod/qe to secret server before merging

**Comment by anshupande** [comment]
secrets are saved and I will merge this now
